### ISSUE TITLE
Swapchain out of date repeatedly on colour-uniform example on Metal

### ISSUE THREAD

**Issue by Wojtechnology** [gfx-rs/gfx#3127]
Short info header:
- GFX version: 0.4.1 (commit: f999b5295ca30dc38f33b0a4770a31d50138687c)
- OS: macOS High Sierra/Mojave
- GPU: NVIDIA GeForce GT 750M 2 GB/Intel UHD Graphics 630 1536 MB

When running the colour-uniform example on multiple Macbook Pros, I noticed a lot of swapchain recreated events triggered by: "Swapchain is out of date and needs to be re-created" error when presenting. This happens when I am not resizing the window. Also, this doesn't happen for the quad example. Anyone have any ideas about what might be going on?

**Comment by kvark** [comment]
Yeah, the quad example is using the new swapchain model (that is not like Vulkan) designed to work nice on everything (but Vulkan). It's also recommended for the users. The old Vulkan-like model is only here for Vulkan Portability, also possibly Amethyst.

That is to say, your experience is still a bug for us.
